hw/ip/otbn/dv/rig/rig/gens/untaken_branch.py

```diff
@@ -0,0 +1,133 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import random
+from typing import Optional
+
+from shared.operand import ImmOperandType, RegOperandType
+from shared.insn_yaml import InsnsFile
+
+from ..config import Config
+from ..model import Model
+from ..program import ProgInsn, Program
+
+from ..snippet import ProgSnippet
+from ..snippet_gen import GenCont, GenRet, SnippetGen
+
+
+class UntakenBranch(SnippetGen):
+    '''A snippet generator that generates program with aligned or
+
+    unalligned addresses for which branches are not taken.
+
+    '''
+
+    def __init__(self, cfg: Config, insns_file: InsnsFile) -> None:
+        super().__init__()
```

Comment:
I wonder whether it would make sense to have some `BranchGen` base class that `BadBranch` and `UntakenBranch` both derive from? That would avoid us having to duplicate the code here.

If you don't fancy doing that as part of this PR, that's fine. In which case, could you open an issue tracking the job?